lib/valueflow.cpp

```diff
@@ -333,6 +335,72 @@ static bool isComputableValue(const Token* parent, const ValueFlow::Value& value
     return true;
 }
 
+uint32_t convertToMultiCharInt(const std::string& x)
+{
+    assert(x.size() < 5);
+    uint32_t y = 0;
+    int shift = (x.size() * 8) - 8;
+    for (size_t i = 0; i < x.size(); ++i) {
+        y |= (uint32_t(uint8_t(x[i])) << shift);
+        shift -= 8;
+    }
+    return y;
+}
+
+template <class T>
+T asInt(T x)
+{
+    return x;
+}
+
+MathLib::bigint asInt(float x) { return x; }
+
+MathLib::bigint asInt(double x) { return x; }
+
+template <class T, class U>
+static T calculate(const std::string& s, T x, U y)
+{
+    switch (convertToMultiCharInt(s)) {
+    case '+':
+        return x + y;
+    case '-':
+        return x - y;
+    case '*':
+        return x * y;
+    case '/':
+        return x / y;
+    case '%':
+        return asInt(x) % asInt(y);
+    case '&':
+        return asInt(x) & asInt(y);
+    case '|':
+        return asInt(x) | asInt(y);
+    case '^':
+        return asInt(x) ^ asInt(y);
+    case '>':
+        return x > y;
+    case '<':
+        return x < y;
+    case '<<':
+        return asInt(x) << asInt(y);
+    case '>>':
+        return asInt(x) >> asInt(y);
+    case '&&':
+        return x && y;
+    case '||':
+        return x || y;
+    case '==':
+        return x == y;
+    case '!=':
+        return x != y;
+    case '>=':
+        return x >= y;
+    case '<=':
+        return x <= y;
+    }
+    throw std::runtime_error("Unknown operator: " + s);
```

Comment:
Dont we catch the exceptions? When there is a syntax error we throw an exception, and the frontend catches it and reports it back to the user.

We should never reach this point and if we do it would be a major bug, I could change it to `assert(false)` so we catch the error during debug.

Reply:
> We should never reach this point and if we do it would be a major bug, I could change it to assert(false) so we catch the error during debug.

Please change it to some `InternalError`.. yes that would stop all analysis of the file which is a bit unfortunate perhaps.. but it would be good if the daca/users will see this bug.